Entities/Natural/Trees/AppleTreeLogic.as

```diff
@@ -0,0 +1,384 @@
+				{
+					scalex = -1.0f;
+				}
+
+				for (int spriteindex = 0; spriteindex < 3; spriteindex++)
+				{
+					CSpriteLayer@ newsegment = this.addSpriteLayer("leaves " + i + " " + spriteindex, "Entities/Natural/Trees/FruitTrees.png" , 32, 32, 0, 0);
+
+					if (newsegment !is null)
+					{
+						Animation@ animGrow = newsegment.addAnimation("grow", 0, false);
+						float z = -550.0f - (vars.height * 10.0f);
+
+						if (spriteindex == 0)
+						{
+							animGrow.AddFrame(11);
+						}
+						else
+						{
+							animGrow.AddFrame(12);
+							z =  500.0f + (vars.height * 10.0f);
+						}
+
+						newsegment.SetAnimation(animGrow);
+						newsegment.ResetTransform();
+						newsegment.SetRelativeZ(z);
+						newsegment.RotateBy(segment.angle, Vec2f(0, 0));
+						Vec2f offset = Vec2f(16 * scalex, 4);
+
+						if (spriteindex == 1)
+						{
+							offset = Vec2f(-16 * scalex, -4);
+						}
+						else if (spriteindex == 2)
+						{
+							offset = Vec2f(22 * scalex, -12);
+						}
+
+						newsegment.TranslateBy(segment.start_pos + offset);
+					}
+				}
+
+				{
+					CSpriteLayer@ newsegment = this.addSpriteLayer("leaves " + i + " " + 3, "Entities/Natural/Trees/FruitTrees.png" , 64, 32, 0, 0);
+
+					if (newsegment !is null)
+					{
+						Animation@ animGrow = newsegment.addAnimation("grow", 0, false);
+						animGrow.AddFrame(3);
+						newsegment.SetAnimation(animGrow);
+						newsegment.ResetTransform();
+						newsegment.SetRelativeZ(550.1f);
+						newsegment.RotateBy(segment.angle, Vec2f(0, 0));
+						newsegment.TranslateBy(segment.end_pos);
+					}
+				}
+
+				{
+					CSpriteLayer@ newsegment = this.addSpriteLayer("leaves " + i + " " + 4, "Entities/Natural/Trees/FruitTrees.png" , 64, 32, 0, 0);
+
+					if (newsegment !is null)
+					{
+						Animation@ animGrow = newsegment.addAnimation("grow", 0, false);
+						animGrow.AddFrame(7);
+						newsegment.SetAnimation(animGrow);
+						newsegment.ResetTransform();
+						newsegment.SetRelativeZ(-550.0f - (i * 5.0f));
+						newsegment.RotateBy(segment.angle, Vec2f(0, 0));
+						newsegment.TranslateBy(segment.end_pos);
+					}
+				}
+				// Time to produce!
+				GrowApples(blob);
+			}
+
+			if (segment.grown_times < 5)
+			{
+				CSpriteLayer@ segmentlayer = this.getSpriteLayer("segment " + i);
+
+				if (segmentlayer !is null)
+				{
+					segmentlayer.animation.frame++;
+
+					if (i == vars.max_height - 1 && segment.grown_times == 3)
+					{
+						segmentlayer.SetOffset(segmentlayer.getOffset() + Vec2f(8, 0));
+					}
+				}
+			}
+		}
+	}
+}
+
+void onDie(CBlob@ this)
+{
+	// Drop apples on death if they still exist
+	// Apple 1
+	CBlob@ apple1 = getBlobByNetworkID(this.get_u16("apple1"));
+    if (apple1 !is null)
+    {            
+        CShape@ shape1 = apple1.getShape();
+        if (shape1 !is null)
+        {
+            shape1.SetStatic(false);
+        }
+    }
+	// Apple 2
+	CBlob@ apple2 = getBlobByNetworkID(this.get_u16("apple2"));

```

Comment:
Once again you're doing the same thing twice. Move this to a for loop and for each apple do `CBlob@ apple = getBlobByNetworkID(this.get_u16("apple" + i));`.

Reply:
Done.